### ISSUE TITLE
Picker: The screen reader does not read correctly all options

### ISSUE THREAD

**Issue by marcoscv-work** [liferay/clay#5281]
We're not sure if we're doing something wrong, the question was asked on slack:
https://liferay.slack.com/archives/C04CPMRMXAL/p1672242567217899

The screen reader does not read correctly all the options.
In each Option it just repeat the name of the first one (maybe because it is the selected key).

You can find an example of this here: https://codesandbox.io/s/experience-selector-picker-example-forked-smgndp?file=/src/index.js:2225-2236

An the PR with it here:
https://github.com/liferay-tango/liferay-portal/pull/3134

**Comment by marcoscv** [comment]
I think it is reproducible in a simple picker with a selected option, attaching video:

Testing url: https://storybook.clayui.com/?path=/story/design-system-components-picker--custom-trigger

https://user-images.githubusercontent.com/6513371/210224643-c9479d20-b511-4658-a3b0-f68e03d56c2c.mov

Changing from Question to Issue

**Comment by matuzalemsteles** [comment]
I did some more tests and implemented the manual announcement to try to solve this but without much success, it seems to be some bug in VoiceOver not announcing correctly, for some reason when there is an option selected and when navigating through the options it announces that the item that is selected is clickable skipping the item announcement with the focus, which even appears in the VoiceOver log, I'm not sure why this is happening but I'm doing more tests if we can avoid this VoiceOver bug.

**Comment by marcoscv-work** [comment]
We can wait with this issue and move it to the back fire. Just we will keep it opened

**Comment by matuzalemsteles** [comment]
Yeah, I'm going to run some more tests today, if that doesn't work, I'll put it on hold.

**Comment by matuzalemsteles** [comment]
I think I found the problem. We have two issues here, it seems to be a Chrome or VoiceOver bug, we need to test this in NVDA and JAWS to see if this is a VoiceOver or Chrome bug.

- The first is that we use the `aria-controls` and `aria-owns` attributes to relate the trigger to the menu when both are not in the same tree, this means that `aria-activedescendant` should work correctly the bug is that when we have an item selected when browsing the list it will always announce that the selected item is clickable, original bug of the issue.
- The second is that just defining the `aria-controls` it has some behaviors in the screen reader announcement, when the menu is opened VoiceOver does not announce the item with the visual focus but when navigating through the menu it announces correctly. This appears to be a VoiceOver bug.
- Setting only `aria-owns` goes back to the first problem.

In theory `aria-controls` and `aria-owns` would have the same behaviors we want to create the relationship but both seem to have SR bug, for now I've only tested it with VoiceOver, I'll test it next week with JAWS and NVDA to make sure it's a VoiceOver bug.

### Possible solutions

A solution I found without having to do a lot of implementation would be to use the trigger as `input` with `aria-controls` it doesn't seem to have any VoiceOver bugs in any of the above cases that I tested but the problem is that in the use cases that we have the trigger it can have icons and any other element that isn't just text, so I'm discarding this solution.

Another option would be instead of using the visual focus, using the browser's focus, but I don't know if the use of `aria-activedescendant` is really important for this case, maybe this is not a problem because our trigger is not a input where the user could write. This would also make the implementation a little simpler.

The next step we can take to resolve this issue once we identify it as a VoiceOver bug is to build a live announcer that will be used as a fallback on Apple devices (Mac, IOS, iPad) that use VoiceOver to correctly announce the behavior of the item when opening the menu, maybe we don't need it when you move the visual focus through items in the menu because VoiceOver correctly announces even when the item is selected that it's a bug in `aria-owns`. One problem I encountered is that after announcing the item with visual focus on opening it keeps reading the trigger focus description which can be confusing for the SR user. I will investigate if I could avoid this next week.

Some final thoughts, even if we consider the second option of using focus instead of `aria-activedescendant` for the visual focus we will still have to correct and test that this will not have bugs in Autocomplete where we will need this behavior because it is a combobox.

**Comment by matuzalemsteles** [comment]
Today I did some more tests now using JAWS, it seems to have a bug with you only having `aria-controls` on Windows but it works great when you have `aria-owns`. So it seems that really the bug may be in VoiceOver that affects Apple devices, whether on Mac or iPhone.

The strategy I'm going to follow here is still trying to keep the visual cursor with `aria-activedescendant` to follow the w3c recommendations without using a different strategy but to fix this with VoiceOver I'm going to follow this strategy:

- Apple devices, we will remove `aria-owns` and only use `aria-controls` for other devices we will keep `aria-owns` which works great with JAWS
- Write a LiveAnnouncer implementation to announce when the Picker menu opens, no need to announce when the item receives visual focus this works fine with VoiceOver.

This implementation will only be for Apple devices.
